main.py

```diff
@@ -27,86 +32,147 @@
     password=None
 )
 
+class PrintJob(Enum):
+    COMPLETED = 0
+    PENDING = 1
+    FILE_ERROR = 2
+    QUOTA_LIMIT = 3
+    JOB_ERROR = 4
+
+print_jobs = {name: {} for name in CONFIG.PRINTERS.NAMES}
+
+class Job():
+    def cleanup():
+        global print_jobs
+        def notdone(job):
+            status = job.current_status()
+            if status == PrintJob.COMPLETED:
+                return job.last_updated + CONFIG.PERSIST_TIME.COMPLETED > time.time()
+            elif status == PrintJob.PENDING:
+                return job.last_updated + CONFIG.PERSIST_TIME.DEFAULT > time.time()
+            return job.last_updated + CONFIG.PERSIST_TIME.ERROR > time.time()
+        
+        print_jobs = {
+            printer: {
+                id: job for id, job in jobs.items() if notdone(job)
+            } for printer, jobs in print_jobs.items()
+        }
+
+    def get_recent(time):
+        jobs = {}
+        for printer_name, printer in print_jobs.items():
+            jobs[printer_name] = [job.toJSON() for job_id, job in printer.items() if job.last_updated > time]
+        return jobs
+
+    def add(printer_name, username, time, status, job_id):
+        if job_id in print_jobs[printer_name]:
+            print_jobs[printer_name][job_id].update(username, status, time)
+        else:
+            print_jobs[printer_name][job_id] = Job(username, time, status, job_id)
+        Job.cleanup()
+
+    def __init__(self, username, time, status, job_id):
+        self.username = username
+        self.id = job_id
+        self.last_updated = time
+        self.status_queue = [(status, time)]
+
+    def update(self, username, status, time):
+        if username != self.username:
+            print("Conflict: Job#" + job_id + " | " +
+                  self.username + " vs " + username)
+            return False
+        self.status_queue.append((status, time))
+        self.last_updated = time
+        return True
+
+    def current_status(self):
+        return self.status_queue[-1][0]
+
+    def toJSON(self):
+        temp = {
```

Comment:
Technically this is not "toJSON". We're simply wrapping it in a Python object that we plan to jsonify later. I recommend either mentioning this or naming it something more appropriate.